src/core/thread/mle.cpp

```diff
@@ -3008,7 +3008,17 @@ otError Mle::HandleAnnounce(const Message &aMessage, const Ip6::MessageInfo &aMe
     }
     else
     {
-        SendAnnounce(static_cast<uint8_t>(channel.GetChannel()), false);
+        if (localTimestamp->Compare(timestamp) < 0)
```

Comment:
nit: prefer to inline this into a single `else if` above to avoid additional indent.